ssg/controls.py

```diff
@@ -183,7 +183,14 @@ def add_references(self, reference_type, rules):
             rule = rules.get(selection)
             if not rule:
                 continue
-            rule.add_extra_reference(reference_type, self.id)
+            try:
+                rule.add_extra_reference(reference_type, self.id)
+            except Exception as exc:
+                msg = (
+                    "Please remove any duplicate listing of rule '%s' in "
+                    "control '%s'." % (
+                        rule.id_, self.id))
+                raise ValueError(msg)
```

Comment:
Instead of creating a new ValueError exception, can we reuse the one we're expecting (should be saved already in `exc`)?

Reply:
@rhmdnd I'm not sure how the exception should be reused, do you have an example?

Originally, I had done `raise ValueError(msg) from exc`, which indicated that one exception was the direct cause of the other, but python2 doesn't support it.
https://github.com/ComplianceAsCode/content/pull/11593/commits/27ff094adf5b1e507190674fe55a8880fe3fdb4e

Current traceback is:
```
Traceback (most recent call last):
  File "/home/wsato/git/content/ssg/controls.py", line 187, in add_references
    rule.add_extra_reference(reference_type, self.id)
  File "/home/wsato/git/content/ssg/build_yaml.py", line 1016, in add_extra_reference
    raise ValueError(msg)
ValueError: Rule ocp_insecure_registries already contains a 'stigid' reference with value 'CNTR-OS-000010'.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/wsato/git/content/build-scripts/compile_all.py", line 179, in <module>
    main()
  File "/home/wsato/git/content/build-scripts/compile_all.py", line 167, in main
    controls_manager.add_references(loader.all_rules)
  File "/home/wsato/git/content/ssg/controls.py", line 524, in add_references
    policy.add_references(rules)
  File "/home/wsato/git/content/ssg/controls.py", line 420, in add_references
    control.add_references(self.reference_type, rules)
  File "/home/wsato/git/content/ssg/controls.py", line 193, in add_references
    raise ValueError(msg)
ValueError: Please remove any duplicate listing of rule 'ocp_insecure_registries' in control 'CNTR-OS-000010'.
[6/26] [rhcos4-content] compiling everything
ninja: build stopped: subcommand failed.
```

Reply:
One reason I re raise the exception is to list in which control the problem is.
When `rule.add_extra_reference()` raises the exception it doesn't know in which control the problem.